attestation-service/attestation-service/src/token/simple.rs

```diff
@@ -5,42 +5,82 @@
 use anyhow::*;
 use base64::engine::general_purpose::URL_SAFE_NO_PAD;
 use base64::Engine;
-use rsa::pkcs1v15::SigningKey;
-use rsa::sha2::Sha384;
-use rsa::signature::{RandomizedSigner, SignatureEncoding};
-use rsa::traits::PublicKeyParts;
-use rsa::RsaPrivateKey;
+use openssl::rsa::Rsa;
+use openssl::sign::Signer;
+use openssl::x509::X509;
+use openssl::{
+    hash::MessageDigest,
+    pkey::{PKey, Private},
+};
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
 use serde_json::{json, Value};
 
 use crate::token::{AttestationTokenBroker, AttestationTokenConfig};
 
-const ISSUER_NAME: &str = "CoCo-Attestation-Service";
-const RSA_KEY_BITS: usize = 2048;
+pub const COCO_AS_ISSUER_NAME: &str = "CoCo-Attestation-Service";
+const RSA_KEY_BITS: u32 = 2048;
 const SIMPLE_TOKEN_ALG: &str = "RS384";
 
 pub struct SimpleAttestationTokenBroker {
-    private_key: RsaPrivateKey,
+    private_key: Rsa<Private>,
     config: AttestationTokenConfig,
+    cert_url: Option<String>,
+    cert_chain: Option<Vec<X509>>,
 }
 
 impl SimpleAttestationTokenBroker {
     pub fn new(config: AttestationTokenConfig) -> Result<Self> {
-        let mut rng = rand::thread_rng();
-        let private_key = RsaPrivateKey::new(&mut rng, RSA_KEY_BITS)?;
+        let (private_key, cert_url, cert_chain) = match config.signer.clone() {
+            Some(signer) => {
+                let pem_data = std::fs::read(signer.key_path)
+                    .map_err(|e| anyhow!("Read Token Signer private key failed: {:?}", e))?;
+                let cert_chain = match signer.cert_path {
+                    Some(cert_path) => {
+                        let pem_cert_chain = std::fs::read_to_string(cert_path)
+                            .map_err(|e| anyhow!("Read Token Signer cert file failed: {:?}", e))?;
+                        let mut chain = Vec::new();
+
+                        for pem in pem_cert_chain.split("-----END CERTIFICATE-----") {
+                            let trimmed = format!("{}\n-----END CERTIFICATE-----", pem.trim());
+                            if !trimmed.starts_with("-----BEGIN CERTIFICATE-----") {
+                                continue;
+                            }
+                            let cert = X509::from_pem(trimmed.as_bytes())
+                                .map_err(|_| anyhow!("Invalid PEM certificate chain"))?;
+                            chain.push(cert);
+                        }
+                        Some(chain)
```

Comment:
could you try to use `and_then` to prevent the pattern matching of `None => None`?

Reply:
Here use `match` may be better, because we need return Error description when read cert failed or verify cert failed, if we use `and_then`, report these error detail will be difficult.